### PR TITLE
Fix devnet-up when op-program/bin is missing prestate-proof.json

### PR DESCRIPTION
**Pull request by anacrolix**
I was running into issues with `make devnet-up`.

https://oplabs-pbc.slack.com/archives/CQZ0U9FAN/p1723006697894859

This checks additionally checks that `op-program/bin/prestate-proof.json` is present meaning `cannon-prestate` can be used as a dependency.

If it's determined that other outputs of `generate-cannon-prestate` are needed, those can be added as additional outputs and deps of `generate-cannon-prestate` and `cannon-prestate` respectively.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -160,16 +167,14 @@ nuke: clean devnet-clean ## Completely clean the project directory
 	git clean -Xdf
 .PHONY: nuke
 
-pre-devnet: submodules ## Prepares for running a local devnet
+## Prepares for running a local devnet
+pre-devnet: submodules cannon-prestate
```
Comment (sebastianst) [inline Makefile:171]: Could we also just directly add
```suggestion
pre-devnet: submodules op-program/bin/prestate-proof.json
```
here and then skip the definition of the phony `cannon-prestate` target?

**Review by sebastianst — APPROVED** [review]
I like that this fixes the Makefile in a way how Make was intended to be used, by depending on actual files and their timestamps :)

File: Makefile (hunk 1 above)
Comment (anacrolix) [reply Makefile:171]: I kept cannon-prestate as the phony that wraps all the outputs (yeah there's only one right now but it's not nice to name it from outside the makefile). I think this is better than dropping the phony and leaving documentation and users confused when the phony is very nice and convenient.

**Review by protolambda — APPROVED** [review]

**Comment by ajsutton** [comment]
This breaks `make cannon-prestate` because it only partially declares dependencies so now it doesn't rebuild even if there are changes:

```
 make cannon-prestate
make: Nothing to be done for `cannon-prestate'.
```

but there in fact changes that need to be built into a new prestate.  We either need to declare _all_ dependencies accurately (basically replacing what the go build tools do) or not declare any dependencies.

File: Makefile [hunk 2]
```diff
@@ -137,11 +137,18 @@ reproducible-prestate:   ## Builds reproducible-prestate binary
 	make -C ./op-program reproducible-prestate
 .PHONY: reproducible-prestate
 
-cannon-prestate: op-program cannon ## Generates prestate using cannon and op-program
+# Checks if prestate outputs are missing
+cannon-prestate: op-program/bin/prestate-proof.json
+.PHONY: cannon-prestate
+
+op-program/bin/prestate-proof.json:
```
Comment (ajsutton) [inline Makefile:144]: I think this specifically needs to declare a dependency on every file in op-program and all of its transitive dependencies like the derive package. Right now if op-program/bin/prestate-proof.json exists it assumes its up to date which is wrong.

**Pull request merged**